libs/shared/src/main/java/io/crate/common/collections/Maps.java

```diff
@@ -51,6 +52,40 @@ public static <K, V> Map<K, V> concat(Map<K, V> m1, Map<K, V> m2) {
         return Collections.unmodifiableMap(result);
     }
 
+    /**
+     * Similar to {@link java.util.AbstractMap#equals(Object)}}
+     * but compares nested arrays by values instead of relying on default Array implementation comparing by reference.
+     */
+    public static boolean equals(Map<?, ?> m1, Map<?, ?> m2) {
+        if (m1 == m2) {
+            return true;
+        }
+
+        if (m1.size() != m2.size()) {
+            return false;
+        }
+
+        try {
+            for (Map.Entry<?, ?> e : m1.entrySet()) {
+                var key = e.getKey();
+                var value = e.getValue();
+                var value2 = m2.get(key);
+                if (value == null) {
+                    if (!(value2 == null && m2.containsKey(key)))
+                        return false;
+                } else {
+                    // Compares arrays by values.
+                    if (!Objects.deepEquals(value, value2)) {
+                        return false;
+                    }
```

Comment:
Please note that this still requires arguments be the arrays of the same type. 
Ie we cannot compare `double[], Double[]` it will fall back to identity equals.

That's why I changed `test_cast_wkt_point_string_array_to_geo_shape_array` to have primitive in the assertion - because in the real execution we cannot have different types and `Double[]` usage in the assertion is rather artifical.

Another option could be to get back GeoShape construction and check equality like
`shape1.within(shape2) and shape2.within(shape1)`- but I don't think it's ay better and we even have a TODo to compare without building shapes.